### ISSUE TITLE
JedisExhausedPoolException : Could not get a resource since the pool is exhausted ..

### ISSUE THREAD

**Issue by mjmanav4** [xetorthio/jedis#2098]
This is the code i am using 

`GenericObjectPoolConfig config = new GenericObjectPoolConfig();
        config.setMaxTotal(30); // // maximum active connections
        config.setMaxWaitMillis(2000);
  
       // jedis cluster nodes is the list of nodes in cluster
        JedisCluster jedisCluster = new JedisCluster(jedisClusterNodes,config);
 Map<String, JedisPool> jpool = jedisCluster.getClusterNodes();

   for (Map.Entry<String, JedisPool> entry : jpool.entrySet()) {
Jedis j = entry.getValue().getResource();
  ScanParams scanParams = new ScanParams().count(100).match("*:"+suffix);

   String cur = redis.clients.jedis.ScanParams.SCAN_POINTER_START;
                boolean cycleIsFinished = false;

                while(!cycleIsFinished) {
                   ....... // some processing
                    cur = scanResult.getCursor();
                    if (cur.equals("0")) {
                        cycleIsFinished = true;
                    }
                }
// DO I NEED TO DO THIS ONCE I COMPLETE MY PROCESSING
entry.returnResource(j)
}`

Do i need to return the resource after each execution ?

**Comment by sazzad16** [comment]
@mjmanav4 

> Do i need to return the resource after each execution ?

YES. But instead of `entry.returnResource(j)`, use `j.close();`. It'd be even better if you could do it within shorter period of time. I.e. do both `getResource()` and `close()` in your while loop. Also, you can use try-with-resources which will automatically `close()` for you.

```
try (Jedis j = entry.getValue().getResource()) {
....... // some processing
} // j.close() is not required anymore. It is done automatically.
```

Already commented in https://github.com/xetorthio/jedis/issues/407#issuecomment-555391963 and https://github.com/xetorthio/jedis/issues/2060#issuecomment-555393570

PS: Man, take some time to breathe.
